tutorials/physics/physics_introduction.rst

```diff
@@ -286,19 +286,19 @@ For example, here is the code for an "Asteroids" style spaceship:
         private Vector2 _thrust = new Vector2(0, -250);
         private float _torque = 20000;
 
-        public override void _IntegrateForces(Physics2DDirectBodyState state)
+        public override void _IntegrateForces(PhysicsDirectBodyState2D state)
         {
             if (Input.IsActionPressed("ui_up"))
-                AppliedForce = _thrust.Rotated(Rotation);
+                state.AppliedForce = _thrust.Rotated(Rotation);
             else
-                AppliedForce = new Vector2();
+                state.AppliedForce = new Vector2();
 
             var rotationDir = 0;
             if (Input.IsActionPressed("ui_right"))
                 rotationDir += 1;
             if (Input.IsActionPressed("ui_left"))
                 rotationDir -= 1;
-            AppliedTorque = rotationDir * _torque;
+            state.AppliedTorque = rotationDir * _torque;
```

Comment:
```suggestion
            state.ApplyTorque(rotationDir * _torque);
```

Reply:
You are right. I should have taken more time on including your suggested changes.